### PR TITLE
dump info during deployment that is useful for etherscan verification and general troubleshooting

### PR DESCRIPTION
**Pull request by snd**
currently deployment is hard to troubleshoot

this is a first step into making deployment simpler, more modular and more transparent

follow up by: https://github.com/paritytech/parity-bridge/issues/120

### CODE REVIEW

File: bridge/src/bridge/deploy.rs [hunk 1]
```diff
@@ -116,3 +159,72 @@ impl<T: Transport + Clone> Future for Deploy<T> {
...
+	);
+
+	let constructor_arguments_bytecode = &deployed_bytecode_hex[bytecode_hex.len()..];
+
+	let mut file = File::create(dir.join("constructor_arguments_bytecode"))?;
+	file.write_all(constructor_arguments_bytecode.as_bytes())?;
+
+	File::create(dir.join("abi"))?.write_all(abi.as_bytes())?;
+
+	Ok(())
```
Comment (debris) [inline bridge/src/bridge/deploy.rs]: why are they so many files?

File: bridge/src/bridge/deploy.rs [hunk 2]
```diff
@@ -93,12 +104,44 @@ impl<T: Transport + Clone> Future for Deploy<T> {
...
+					ref home_data,
+					ref foreign_data
+				} => {
 					let (home_receipt, foreign_receipt) = try_ready!(future.poll().map_err(ErrorKind::Web3));
+
+					let home_contract_address = home_receipt.contract_address.expect("contract creation receipt must have an address; qed");
+					let foreign_contract_address = foreign_receipt.contract_address.expect("contract creation receipt must have an address; qed");
+
+					write_deployment_info(
+						format!("deployment-home-{}", home_contract_address.to_hex()),
```
Comment (debris) [inline bridge/src/bridge/deploy.rs]:
```
format!("deployment-home-{:x}", home_contract_address)
```

is shorter and does 1 allocation less ;)

File: bridge/src/bridge/deploy.rs (hunk 1 above)
Comment (snd) [reply bridge/src/bridge/deploy.rs]: i think it's the simplest way to achieve this.

we could merge smaller fields into one json or toml file but that makes it more complicated

File: bridge/src/bridge/deploy.rs (hunk 2 above)
Comment (snd) [reply bridge/src/bridge/deploy.rs]: good to know for the future

File: bridge/src/bridge/deploy.rs (hunk 1 above)
Comment (debris) [reply bridge/src/bridge/deploy.rs]: that's what I thought. I don't have a strong opinion about this and I was just asking to confirm that there is no other reason :)

Comment (snd) [reply bridge/src/bridge/deploy.rs]: the nice thing about this and goal of the change is that we can now just ask someone for the `deploy-{home|foreign}-{tx-hash}` folder and have all the information we could need for troubleshooting : ) the folder also contains all information one needs to use https://etherscan.io/verifyContract

**Pull request merged**